core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/ExcelInputReader.java

```diff
@@ -0,0 +1,457 @@
+  private void init() {
+    if (!Strings.isNullOrEmpty(excelInputreaderConfig.columnList)) {
+      String[] columnsList = excelInputreaderConfig.columnList.split(",");
+      inputColumns = Arrays.asList(columnsList);
+    }
+
+    if (!Strings.isNullOrEmpty(excelInputreaderConfig.columnMapping)) {
+      String[] mappings = excelInputreaderConfig.columnMapping.split(",");
+      for (String map : mappings) {
+        String[] columns = map.split(":");
+        if (CollectionUtils.isNotEmpty(inputColumns) && !inputColumns.contains(columns[0])) {
+          throw new IllegalArgumentException("Column name: " + columns[0] + " in 'Column-Label Mapping' does not " +
+                                               "match the columns in the 'Column To Be Extracted' input text box. " +
+                                               "It has to be one of the columns in 'Column To Be Extracted' " +
+                                               "input text box.");
+        }
+        columnMapping.put(columns[0], columns[1]);
+      }
+    }
+
+    if (!Strings.isNullOrEmpty(excelInputreaderConfig.outputSchema)) {
+      String[] schemaList = excelInputreaderConfig.outputSchema.split(",");
+      for (String schema : schemaList) {
+        String[] columns = schema.split(":");
+        if (CollectionUtils.isNotEmpty(inputColumns) && !inputColumns.contains(columns[0])) {
+          throw new IllegalArgumentException("Column name: " + columns[0] + " in 'Field Name Schema Type Mapping'" +
+                                               " does not match the columns in the 'Column To Be Extracted' input " +
+                                               "text box. It has to be one of the columns in " +
+                                               "'Column To Be Extracted' input text box.");
+        }
+        outputSchemaMapping.put(columns[0], columns[1]);
+      }
+    }
+  }
+
+  @Override
+  public void transform(KeyValue<LongWritable, Object> input, Emitter<StructuredRecord> emitter) throws Exception {
+
+    getOutputSchema();
+    StructuredRecord.Builder builder = StructuredRecord.builder(outputSchema);
+    String inputValue = input.getValue().toString();
+
+    String[] excelRecords = inputValue.split("\t");
+
+    String fileName = excelRecords[1];
+    String sheetName = excelRecords[2];
+    String ifEndRow = excelRecords[3];
+
+    int prevRowNum = Integer.parseInt(excelRecords[0]);
+
+    if (filePrevRowNumMap.containsKey(fileName)) {
+      if (prevRowNum - filePrevRowNumMap.get(fileName) > 1 &&
+        excelInputreaderConfig.terminateIfEmptyRow.equalsIgnoreCase("true")) {
+        throw new ExecutionException("Encountered empty row while reading Excel file :" + fileName +
+                                       " . Terminating processing", new Throwable());
+      }
+    }
+
+    filePrevRowNumMap.put(fileName, prevRowNum);
+
+    Map<String, String> excelColumnValueMap = new HashMap<>();
+
+    for (String columns : excelRecords) {
```

Comment:
If excelRecords has more rows, then it would override excelColumnMap. Down below you use this to write to structured record and it seems wrong. This logic needs a re-write assuming that excelRecords can have more than 1 row in it.